src/java/ognl/AccessibleObjectHandlerJDK9Plus.java

```diff
@@ -0,0 +1,236 @@
+//  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+//  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+//  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+//  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+//  OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+//  AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+//  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+//  THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+//  DAMAGE.
+//--------------------------------------------------------------------------
+package ognl;
+
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+
+/**
+ * Utilizes a JDK 9 and later mechanism for changing the accessibility level of a given
+ *   AccessibleObject.
+ *
+ * If the JDK 9+ mechanism fails, this class will fall back to a standard pre-JDK 9 reflection mechanism.
+ *   Note:  That may cause "WARNING: Illegal reflective access" output to be generated to stdout/stderr.
+ *
+ * For reference, this class draws on information from the following locations:
+ *   - Post about Illegal Reflective Access <a href="https://stackoverflow.com/questions/50251798/what-is-an-illegal-reflective-access">what is an illegal reflective access</a>
+ *   - Blog on Unsafe <a href="http://mishadoff.com/blog/java-magic-part-4-sun-dot-misc-dot-unsafe/">Java Magic. Part 4: sun.misc.Unsafe</a>
+ *   - Blog on Unsafe <a href="https://www.baeldung.com/java-unsafe">Guide to sun.misc.Unsafe</a>
+ *   - JEP about access to Unsafe being retained in JDK 9 <a href="https://openjdk.java.net/jeps/260">JEP 260: Encapsulate Most Internal APIs</a>
+ *
+ * In addition to the above, inspiration was drawn from Gson: <a href="https://github.com/google/gson/pull/1218">PR 1218</a>,
+ *   <a href="https://github.com/google/gson/pull/1306">PR 1306</a>.
+ *
+ * Appreciation and credit to the authors, contributors and commenters for the information contained in the preceding links.
+ *
+ * @since 3.1.24
+ */
+class AccessibleObjectHandlerJDK9Plus implements AccessibleObjectHandler
+{
+    private static final Class _clazzUnsafe = instantiateClazzUnsafe();
+    private static final Object _unsafeInstance = instantiateUnsafeInstance(_clazzUnsafe);
+    private static final Method _unsafeObjectFieldOffsetMethod = instantiateUnsafeObjectFieldOffsetMethod(_clazzUnsafe);
+    private static final Method _unsafePutBooleanMethod = instantiateUnsafePutBooleanMethod(_clazzUnsafe);
+    private static final Field _accessibleObjectOverrideField = instantiateAccessibleObjectOverrideField();
+    private static final long _accessibleObjectOverrideFieldOffset = determineAccessibleObjectOverrideFieldOffset();
+
+    /**
+     * Package-accessible constructor
+     */
+    AccessibleObjectHandlerJDK9Plus() {}
```

Comment:
Why not private?

Reply:
That was intended to restrict creation of instances to the package, so that only classes within the main package could construct an instance.
With the latest change the constructors are now private and instead a package-level generator method is called to produce an instance (which serves the same purpose, but is a bit easier to read ... plus they just return the generic type reference).